kernel/src/memory/physical_allocators/monotonic_allocator.hpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023 Eliyahu Gluschove-Koppel.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef HUGOS_KERNEL_SRC_MEMORY_PHYSICAL_ALLOCATORS_MONOTONIC_ALLOCATOR_HPP
+#define HUGOS_KERNEL_SRC_MEMORY_PHYSICAL_ALLOCATORS_MONOTONIC_ALLOCATOR_HPP
+
+#include "../physical_allocator.hpp"
+#include "../types.hpp"
+
+#include <multiboot/memory_map.hpp>
+#include <stdint.h>
+
+namespace memory::physical_allocators {
+	/**
+	 * Add stuff
+	 */
+	class MonotonicAllocator : public IPhysicalAllocator {
+	private:
+		aligned<paddr_t> next_frame;
+		aligned<paddr_t> end_frame;
+		aligned<paddr_t> kernel_start_frame;
+		aligned<paddr_t> kernel_end_frame;
+		aligned<paddr_t> multiboot_start_frame;
+		aligned<paddr_t> multiboot_end_frame;
+		multiboot::tags::MemoryMap *mem_map;
+
+	public:
+		MonotonicAllocator(paddr_t start_addr,
+		                   paddr_t end_addr,
+		                   paddr_t kernel_start,
+		                   paddr_t kernel_end,
+		                   paddr_t multiboot_start,
+		                   paddr_t multiboot_end,
+		                   multiboot::tags::MemoryMap *mem_map) noexcept :
+			next_frame(aligned<paddr_t>::aligned_up(start_addr)),
+			end_frame(aligned<paddr_t>::aligned_up(end_addr)),
+			kernel_start_frame(aligned<paddr_t>::aligned_down(kernel_start)),
+			kernel_end_frame(aligned<paddr_t>::aligned_up(kernel_end)),
+			multiboot_start_frame(aligned<paddr_t>::aligned_down(multiboot_start)),
+			multiboot_end_frame(aligned<paddr_t>::aligned_up(multiboot_end)),
+			mem_map(mem_map) {}
+		MonotonicAllocator(MonotonicAllocator&)             = delete;
+		MonotonicAllocator(MonotonicAllocator&&) noexcept   = default;
+		MonotonicAllocator& operator=(MonotonicAllocator&)  = delete;
```

Comment:
Correct copy assign syntax
```suggestion
		MonotonicAllocator& operator=(const MonotonicAllocator&)  = delete;
```